WordPress/src/main/java/org/wordpress/android/ui/media/services/MediaUploadService.java

```diff
@@ -219,31 +269,31 @@ private void unpackIntent(@NonNull Intent intent) {
         }
     }
 
-    private boolean matchesInProgressMedia(final @NonNull MediaModel media) {
-        return mCurrentUpload != null && media.getLocalSiteId() == mCurrentUpload.getLocalSiteId();
-    }
-
-    private void cancelUpload() {
-        if (mCurrentUpload != null) {
-            dispatchCancelAction(mCurrentUpload);
+    private void cancelUpload(MediaModel oneUpload) {
+        if (oneUpload != null) {
+            dispatchCancelAction(oneUpload, mSiteStore.getSiteByLocalId(oneUpload.getLocalSiteId()));
         }
     }
 
-    private void dispatchUploadAction(@NonNull final MediaModel media) {
-        AppLog.i(AppLog.T.MEDIA, "Dispatching upload action for media with local id: " + media.getId() +
-                " and path: " + media.getFilePath());
-        media.setUploadState(UploadState.UPLOADING.name());
-        mDispatcher.dispatch(MediaActionBuilder.newUpdateMediaAction(media));
+    private void dispatchUploadAction(@NonNull final MediaModel media, @NonNull final SiteModel site) {
+        if (media != null && site != null) {
```

Comment:
Adding these null checks on top of the existing `@NonNull` annotations generates lint warnings. Additionally, it still leaves lint warnings where these methods are called without null checking. In this case, [here](https://github.com/wordpress-mobile/WordPress-Android/pull/5383/files#diff-339e1ea3210c67f894a26e04d833f5f9R192).

I think we should remove these null checks and let the annotation guide us to earlier points where we should check for unexpected `null`s and log them (we're already handling them except in two straightforward cases).